### PR TITLE
Add doc for database auto-backup

### PR DESCRIPTION
**Pull request by hossainemruz**

### CODE REVIEW

File: docs/guides/latest/auto-backup/database.md [hunk 1]
```diff
@@ -11,3 +11,584 @@ product_name: stash
...
+kind: BackupConfigurationTemplate
+metadata:
+  name: pg-backup-template
+spec:
+  # ============== Template for Repository ==========================
+  backend:
+    gcs:
+      bucket: appscode-qa
+      prefix: stash-backup/${TARGET_NAMESPACE}/${TARGET_KIND}/${TARGET_NAME}
+    storageSecretName: gcs-secret
```
Comment (tamalsaha) [inline docs/guides/latest/auto-backup/database.md]: What is the namespace?

Comment (hossainemruz) [reply docs/guides/latest/auto-backup/database.md]: Namespace of the database. Multiple namespace can have database with same name. So, this Namespace variable in path differentiate them.

Comment (tamalsaha) [reply docs/guides/latest/auto-backup/database.md]: So, user has to create the storageSecret in the database namespace ahead of time?

File: docs/guides/latest/auto-backup/database.md [hunk 2]
```diff
@@ -11,3 +11,584 @@ product_name: stash
...
+  # ============== Template for BackupConfiguration =================
+  task:
+    name: pg-backup-11.2
```
Comment (tamalsaha) [inline docs/guides/latest/auto-backup/database.md]: I think that task field should be templatized. I think we should add a `Version` field in AppBinding. https://github.com/kmodules/custom-resources/blob/master/apis/appcatalog/v1alpha1/appbinding_types.go#L28

Then we will be able to use the same BCT for any version of Postgres database.

Comment (hossainemruz) [reply docs/guides/latest/auto-backup/database.md]: That make sense.

Comment (tamalsaha) [reply docs/guides/latest/auto-backup/database.md]: Added Version field. https://github.com/kmodules/custom-resources/pull/21

File: docs/guides/latest/auto-backup/database.md (hunk 1 above)
Comment (hossainemruz) [reply docs/guides/latest/auto-backup/database.md]: yes

Comment (hossainemruz) [reply docs/guides/latest/auto-backup/database.md]: I think, I have missed that to mention in doc.

File: docs/examples/guides/latest/auto-backup/database/backupconfiguration_template.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ spec:
     storageSecretName: gcs-secret
   # ============== Template for BackupConfiguration =================
   task:
-    name: pg-backup-11.2
+    name: pg-backup-${TARGET_APP_VERSION}
```
Comment (tamalsaha) [inline docs/examples/guides/latest/auto-backup/database/backupconfiguration_template.yaml]: I think we should rename these tasks as `postgres-backup-${TARGET_APP_VERSION}`

File: docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupBlueprint
+metadata:
+  name: pg-backup-blueprint
+spec:
+  # ============== Template for Repository ==========================
+  backend:
+    gcs:
+      bucket: appscode-qa
+      prefix: stash-backup/${TARGET_NAMESPACE}/${TARGET_APP_GROUP}/${TARGET_APP_RESOURCE}/${TARGET_NAME}
```
Comment (tamalsaha) [inline docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml]: Then why not just use `${TARGET_APP_TYPE}` ?

File: docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    storageSecretName: gcs-secret
+  # ============== Template for BackupConfiguration =================
+  task:
+    name: pg-backup-${TARGET_APP_VERSION}
```
Comment (tamalsaha) [inline docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml]: Open a pr to rename this to `postgres-`

File: docs/guides/latest/auto-backup/database.md [hunk 6]
```diff
@@ -11,3 +11,604 @@ product_name: stash
...
+backupblueprint.stash.appscode.com/pg-backup-blueprint created
+```
+
+Now, automatic backup is configured for PostgreSQL database. We just have to add a annotation to the `AppBinding` of the targeted database.
+
+> Note: `BackupBlueprint` is a non-namespaced crd. So, you can use a `BackupBlueprint` to backup targets in multiple namespaces. However, Storage Secret is a namespaced object. So, you have to manually create the secret in each namespace where you have a target for backup. Please give us your feedback on how to improve the ux of this aspect of Stash on [GitHub](https://github.com/stashed/stash/issues/842).
+
+**Required Annotation for Auto-Backup Database:**
+
+You have to add the following annotation to the `AppBinding` crd of the targeted database to enable backup for it:
```
Comment (tamalsaha) [inline docs/guides/latest/auto-backup/database.md]: This label will not stick given how we generate labels for KubeDB appbindings https://github.com/kubedb/postgres/blob/master/pkg/controller/appbinding.go#L30

This should be applied to the Database CRD and passed to AppBinding.

cc: @the-redback . Let's talk tomorrow.

File: docs/guides/latest/auto-backup/database.md [hunk 7]
```diff
@@ -11,3 +11,604 @@ product_name: stash
...
+      from: POSTGRES_USER
+      to: username
+  - renameKey:
+      from: POSTGRES_PASSWORD
+      to: password
+  type: kubedb.com/postgres
+  version: "11.2"
+```
+
+> If you have deployed the database without KubeDB, you have to create the AppBinding crd manually.
```
Comment (tamalsaha) [inline docs/guides/latest/auto-backup/database.md]: We need to point to a place that shows how to do that manually.

File: docs/guides/latest/auto-backup/database.md (hunk 6 above)
Comment (tamalsaha) [reply docs/guides/latest/auto-backup/database.md]: So, we forward labels, not annotations. So, this should be fine. 

TODO: Stop setting annotations on AppBinding in KubeDB.

File: docs/examples/guides/latest/auto-backup/database/backupconfiguration_template.yaml (hunk 3 above)
Comment (hossainemruz) [reply docs/examples/guides/latest/auto-backup/database/backupconfiguration_template.yaml]: Fixed

File: docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml (hunk 5 above)
Comment (hossainemruz) [reply docs/examples/guides/latest/auto-backup/database/backupblueprint.yaml]: fixed

File: docs/guides/latest/auto-backup/database.md (hunk 6 above)
Comment (hossainemruz) [reply docs/guides/latest/auto-backup/database.md]: done

**Pull request merged**